nntrainer/src/tensor_dim.cpp

```diff
@@ -61,7 +71,7 @@ int TensorDim::setTensorDim(std::string input_shape) {
     std::sregex_iterator(input_shape.begin(), input_shape.end(), words_regex);
   auto words_end = std::sregex_iterator();
   int cur_dim = std::distance(words_begin, words_end);
-  if (cur_dim > MAXDIM) {
+  if (cur_dim <= 0 || cur_dim > MAXDIM) {
     ml_loge("Tensor Dimension should be less than 4");
```

Comment:
```suggestion
    ml_loge("Tensor Dimension should be between 1 and 4");
```

Reply:
Slightly out of topic though, I think line 80~83 / 85~86 can be safely deleted.

Reply:
Done.
Removed both checks and update `stoi` to `stoul` on line 80.